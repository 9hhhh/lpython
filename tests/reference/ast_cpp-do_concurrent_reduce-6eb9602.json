{
    "basename": "ast_cpp-do_concurrent_reduce-6eb9602",
    "cmd": "cpptranslate --show-ast-cpp {infile}",
    "infile": "tests/do_concurrent_reduce.f90",
    "infile_hash": "44ebdebb05b853f4680c203c3b2b0e98e9285d1f3923ecbc255bc105",
    "outfile": null,
    "outfile_hash": null,
    "stdout": "ast_cpp-do_concurrent_reduce-6eb9602.stdout",
    "stdout_hash": "dfc00a2994c4759b08d7d5a06cb2c1ebc1e0b02b8e6616581f24c53b",
    "stderr": null,
    "stderr_hash": null,
    "returncode": 0
}