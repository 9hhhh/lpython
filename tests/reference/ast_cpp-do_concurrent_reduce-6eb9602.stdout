void sum_reduce(const Kokkos::View<const float*> & a, 
float *s)
{
    size_t N;
    size_t i;
    N = a.extent(0);
    *s = 0
    Kokkos::parallel_reduce(N, KOKKOS_LAMBDA(const long i, float & updatevar_0) {
        updatevar_0 = (updatevar_0) + (a[i]);
    }, *s);
}

